### DISCUSSION TITLE
Automatic promotion of filed int values to float?

### DISCUSSION THREAD

**Discussion by tksharpless** [nlohmann/json#4254] [Q&A]
I have a lot of Json files where all  zero values appear as simply '0', and some float values that happen to be integers are printed as such.    I know which names are really integers and which really floats, and would like to parse them into variables of appropriate types, assigning  a default value when the name is absent.
The value() function seems good for that but unfortunately, when the filed number looks like an int and the default value is float, it fails with a runtime type exception. 
Is there an easy workaround?

**Comment by nlohmann** [comment]
No, unfortunately not.

**Reply by nlohmann** [reply]
You would need to use `is_number_float` (https://json.nlohmann.me/api/basic_json/is_number_float/) and such to check first.

**Comment by tksharpless** [comment]
Are you saying there are no options for type coercion or automatic
promotion?  I hope you will consider offering such options in future
releases, as they could be really useful.  For now, I will write
conditional code that checks the numeric type, as you suggest; but I think
you should reconsider whether runtime type checking really adds any value.
Regards,
Thomas
